vignettes/geohashTools.Rmd

```diff
@@ -0,0 +1,155 @@
+
+The `gh_neighbors` function is designed to address this. Calling `gh_neighbours` will return all of the geohashes adjacent to a given geohash (or vector of geohashes) at the same level of precision. 
+
+For example, the Merlion statue in Singapore is roughly at `w21z74nz`, but this level of precision zooms in a bit too far. The geohash neighborhood thereof can be found with:
+
+```{r neighbors}
+gh_neighbors('w21z74nz')
+```
+
+## Working with spatial formats
+
+There are several `geohashTools` helper functions for converting geohashes into `sp` and `sf` class objects.
+
+### gh_to_*
+
+The `gh_to_sp`, `gh_to_spdf` and `gh_to_sf` functions convert geohash or vector of geohashes into spatial objects of `sp` and `sf` class respectively. Consider the previous example with the Singapore Merlion and some random data.
+```{r merlion_nbhd, fig.width = 6, fig.height = 6, out.width = '\\textwidth'}
+library(sf)
+merlion_ghs <- gh_neighbors("w21z74")
+merlion_nbhd <- gh_to_sf(merlion_ghs)
+plot(merlion_nbhd)
+```
+
+### gh_covering
+
+Sometimes we have a set of spatial datapoints that we want to aggregate or index using geohashes. The `gh_covering` function produces a grid of geohashes that overlap with the spatial points. For this function to work, the spatial object has to be in the [WGS84 (EPSG 4326)](https://epsg.io/4326) coordinate reference system that the geohashes use. 
+
+Let's use the included `meuse` dataset in the `sp` package which shows point coordinates for metal deposits. The first element in this object shows values for cadmium deposits. 
+```{r meuse, fig.width = 6, fig.height = 6, out.width = '\\textwidth'}
+library(ggplot2)
+
+data(meuse, package = "sp")
+meuse_sf = st_as_sf(meuse, coords = c("x", "y"), crs = 28992, agr = "constant")
+meuse_sf <- st_transform(meuse_sf, crs = st_crs(4326))
+
+ggplot() +
+  geom_sf(data = meuse_sf, aes(colour = cadmium)) +
+  theme_void()
+```
+By default, `gh_covering` creates a grid that covers the extent of the bounding box for the spatial object. 
+```{r meuse_covering}
+meuse_gh <- gh_covering(meuse_sf)
+```
+
+We can visualize what this looks like with geohashes overlayed on top.
+
+```{r meuse_covering_plot, fig.width = 6, fig.height = 6, out.width = '\\textwidth'}
+ggplot() +
+  geom_sf(data = meuse_sf) +
+  geom_sf(data = meuse_gh, fill  = NA) +
+  geom_sf_text(data = meuse_gh, aes(label = rownames(meuse_gh))) +
+  theme_void()
+```
+
+Alternatively, using `gh_covering` with the parameter `minimal = TRUE` will only create geohashes for intersecting objects. 
+```{r meuse_minimal}
+# Currently doesn't work
+# meuse_gh <- gh_covering(meuse_sf, minimal = TRUE)
```

Comment:
this works now right?

Reply:
Yep, confirmed in [26db6f5](https://github.com/MichaelChirico/geohashTools/pull/33/commits/26db6f575a2914f30e26e81a3f215606c061de67)